### PR TITLE
Make `get_desynced_commands` work with slash command groups

### PR DESCRIPTION
**Pull request by Ratery**
## Summary
Update `get_desynced_commands` function to be able to properly work with slash command groups.
This PR implements recursive slash commands traversing to compare all nested subcommands.

## Checklist
- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by BobDotCom — DISMISSED** [review]

**Review by EmreTech — DISMISSED** [review]

**Comment by krittick** [comment]
Please resolve conflicts. Thanks!

**Comment by BobDotCom** [comment]
Please run the pre-commit hooks for formatting, the lint action failed

File: discord/bot.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
     Union,
 )
 
+import discord
```
Comment (plun1331) [inline discord/bot.py:47]: Why was this import added

**Comment by krittick** [comment]
Please resolve conflicts. Thanks!
